### PR TITLE
Refactor: use typescript

### PR DESCRIPTION
**Pull request by sharafian**
we've run into a lot of issues in the asym-server that could have been prevented via typechecking, so I think it's a good idea to use typescript here.

So far this changes the code to use typescript but not the tests.

The index.ts also uses module.exports in order to maintain compatibility with commonjs style default imports (Required for the connector)

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  settleDelay: number,
+  expiration?: string,
+  cancelAfter?: string,
+  sourceTag?: number,
+  destinationTag?: number,
+  previousAffectingTransactionID: string,
+  previousAffectingTransactionLedgerVersion: number
+}
+
+export type Store = {
```
Comment (sentientwaffle) [inline src/util.ts]: Should this be referenced in `account.js` `store`?

File: src/store-wrapper.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
-class StoreWrapper {
-  constructor (store) {
+import { Store } from './util'
+
+export default class StoreWrapper {
+  private _store: any // TODO: store type
```
Comment (sentientwaffle) [inline src/store-wrapper.ts]: Why isn't this `Store`?

File: src/index.ts [hunk 3]
```diff
@@ -76,21 +121,23 @@ class Plugin extends MiniAccountsPlugin {
     })
   }
 
-  xrpToBase (amount) {
+  xrpToBase (amount: string | BigNumber) {
```
Comment (sentientwaffle) [inline src/index.ts]: Could use `BigNumber.Value` here and in `baseToXrp`

File: src/util.ts (hunk 1 above)
Comment (sharafian) [reply src/util.ts]: the store in account.js is actually a StoreWrapper, but I'm importing that type now

**Pull request merged**